File: LB/PROgram0063.c
/* 

Q63) Write a Program which accepts N numbers from user and Check whether it consist of 11 or not

    Input  :- 6
    Elements :- 85 66 11 80 93 88

    Output :- 1 

*/

#include<stdio.h>
#include<stdlib.h>
#include<stdbool.h>

typedef int BOOL;
#define TRUE 1
#define FALSE 0

bool Check11(int Arr[], int iLength)
{
    int iCount = 0; 

    for(iCount = 0; iCount < iLength; iCount++)
    {
        if(Arr[iCount] == 11)
        {
            return TRUE;
        }

    }

    return FALSE;
    
}

int main()
{
    int iSize, iCnt = 0;
    int * Brr = 0;
    bool bRet = FALSE;

    printf("Enter Number Of Elements : ");
    scanf("%d",&iSize);

    Brr = (int *)malloc(iSize * sizeof(int));

    printf("Please Enter %d Elements\n",iSize);

    for(iCnt = 0; iCnt < iSize; iCnt++)
    {
        printf("Enter %d Element : ",iCnt + 1);
        scanf("%d",&Brr[iCnt]);
    }

    bRet = Check11(Brr,iSize);

    if(bRet == TRUE)
    {
        printf("Array Consists of 11");
    }    
    else
    {
        printf("Array don't have 11");
    }

    free(Brr);

    return 0;
}

File: PM/PROgram0063.py
'''

Q23) Write a program that accepts one number and checks whether it is a perfect number or not

'''

def Perfect(No):
    Sum = 0

    for i in range(1, No):
        if(No % i == 0):
            Sum = Sum + i

    if(Sum == No):
        return True
    else:
        return False

def main():
    Value = int(input("Enter Number : "))

    Ret = Perfect(Value)

    if(Ret == True):
        print("Perfect Number")
    else:
        print("Not a Perfect Number")

if __name__ == "__main__":
    main()